packages/node_modules/@webex/plugin-meetings/test/unit/spec/meetings/index.js

```diff
@@ -0,0 +1,321 @@
+/*!
+ * Copyright (c) 2015-2019 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {assert} from '@ciscospark/test-helper-chai';
+import sinon from '@ciscospark/test-helper-sinon';
+import Events from '@webex/plugin-meetings/src/events';
+import MockSpark from '@ciscospark/test-helper-mock-spark';
+import Device from '@ciscospark/internal-plugin-wdm';
+import Mercury from '@ciscospark/internal-plugin-mercury';
+import Meetings from '@webex/plugin-meetings/src/meetings';
+import PersonalMeetingRoom from '@webex/plugin-meetings/src/personal-meeting-room';
+import Reachability from '@webex/plugin-meetings/src/reachability';
+import MeetingCollection from '@webex/plugin-meetings/src/meetings/collection';
+import MeetingsUtil from '@webex/plugin-meetings/src/meetings/util';
+import Meeting from '@webex/plugin-meetings/src/meeting';
+
+
+describe('plugin-meetings', () => {
+  let spark;
+  describe('meetings', () => {
+    beforeEach(() => {
+      spark = new MockSpark({
+        children: {
+          device: Device,
+          mercury: Mercury,
+          meetings: Meetings
+        }
+      });
+    });
+    it('has a spark instance with a meetings property', () => {
+      assert.exists(spark, 'spark was initialized with children');
+      assert.exists(spark.meetings, 'meetings child was set up on the spark instance');
+    });
+    describe('Public API Contracts', () => {
+      describe('gets', () => {
+        describe('#getReachability', () => {
+          describe('before #setReachability', () => {
+            it('does not get a reachability instance', () => {
+              const reachability = spark.meetings.getReachability();
+              assert.notExists(reachability, 'reachability is undefined because #setReachability has not been called');
+            });
+          });
+          describe('after #setReachability', () => {
+            beforeEach(() => {
+              spark.meetings.setReachability();
+              const reachabilityMocker = spark.meetings.getReachability();
+              sinon.stub(reachabilityMocker, 'gatherReachability').returns(true);
+            });
+            it('gets the reachability data instance from spark.meetings', () => {
+              const reachability = spark.meetings.getReachability();
+              assert.exists(reachability, 'reachability is defined because #setReachability has been called');
+              assert.typeOf(reachability, 'object', 'should be an object');
```

Comment:
I would rather assert that specific methods exist instead of asserting it's an instanceOf or that it's an object. If we make a breaking change to the API contract, this won't help us. In this case, just asserting that there is a .getReachability() is probably enough. Then we will have separate unit tests for that component later down the line

Reply:
yeah, i like the idea of adding assert.exists checks for the public api/event functions